src/main/java/teammates/common/datatransfer/attributes/EntityAttributes.java

```diff
@@ -55,6 +55,21 @@ public void addNonEmptyError(String error, List<String> errors) {
         errors.add(error);
     }
 
+    /**
+     * Custom equals method that is able to handle null objects.
+     * @param first first object to be compared
+     * @param second second object to be compared
+     * @return true of both objects are null or equal, false otherwise.
+     */
+    protected boolean nullableDataEquals(Object first, Object second) {
```

Comment:
Actually, you can use `Objects.equals` from `java.util.Objects`

Reply:
Replaced all usage of `nullableDataEquals` with `Objects.equals`

Reply:
You can remove the method also?